### ISSUE TITLE
DE Setup causes issues in Magento 2.4.3

### ISSUE THREAD

**Issue by bytes-max** [firegento/firegento-magesetup2#199]
Please make sure that you provide as much information as possibl - thank you!

### General

* Module version: v0.4.1
* Magento version: 2.4.3
* PHP version: 7.4

### Issue description
We see this issue accross all our Magento 2.4.3 installations after setting up FireGento as advised from the docs:
`Exception #0 (UnexpectedValueException): Invalid parameter configuration provided for $configPaths argument of Magento\Variable\Model\Source\Variables`
A solution is to remove the argument section from the di.xml in the firegento setup that comes at last. It does not seem to work as expected this way. 
...

### Steps to reproduce

1. be in development mode
2. Install Firegento and run setup for DE config
2. clear all caches
3. try to make an order, proceed in checkout

**Comment by bytes-max** [comment]
To add more details, its actually the element in `etc/di.xml` starting from Line 60. If I remove this manually, the shop works again. However, for now I am patching the file manually via Jenkins, yet I'd love to see that feature being used at all. Did M2.4.3 stopped making usage of these? For what its worth it, the Magento Core Code does make different usage on the function as it appears to be brought to Magento via this DI.

**Comment by sprankhub** [comment]
Thanks for the report @nopenopenope! I could reproduce the issue. The proper fix for 2.4.3 is to update the class name in `etc/di.xml`:

Old:

    <type name="Magento\Variable\Model\Source\Variables">

New:

    <type name="Magento\Variable\Model\Config\Structure\AvailableVariables">

However, I do not know yet how a backwards-compatible solution to this issue could look like as the old definition is needed for < 2.4.3 and the new definition is needed for >= 2.4.3...

**Comment by bytes-max** [comment]
Classical Magento 2 Development... For now I will fix the DI manually in my deployment process, maybe its feasible to have a tag running for M2.4.3? However, I understand its not quite practicable..

**Comment by sprankhub** [comment]
:see_no_evil: 

I really think there is currently no way to do this in a backwards-compatible way without a lot of hacking (maybe the current approach could be replaced with a plugin, which does different things depending on the version).

I indeed consider tagging a `v1.0.0`, which only supports >= 2.4.3.

Thoughts @mautz-et-tong @frostblogNet?

**Comment by Schrank** [comment]
Do it! 

![Picard: Make it so](https://media.giphy.com/media/bKnEnd65zqxfq/giphy.gif)

**Comment by vertic4l** [comment]
That issue caused so much trouble. Spent like 2 hours to figure it out. Copied firegento from vendor into app/code for now to fix the di.xml file

**Comment by mautz-et-tong** [comment]
@sprankhub We should do two new proper releases to divide between >=2.4.3 and <2.4.3 including updated readme files and composer requirements update

**Comment by avstudnitz** [comment]
If you need a quick solution, you can create a workaround with your own module depending on `FireGento_MageSetup`. Using the following in its `di.xml` seems to fix it:

```
<?xml version="1.0"?>
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
    <type name="Magento\Email\Model\Source\Variables">
        <plugin name="mageSetupEmailSourceVariables" disabled="true"/>
    </type>

    <type name="Magento\Variable\Model\Source\Variables">
        <arguments>
            <argument name="configPaths" xsi:type="object">Magento\Variable\Model\Config\Structure\AvailableVariables</argument>
        </arguments>
    </type>

    <type name="Magento\Variable\Model\Config\Structure\AvailableVariables">
        <arguments>
            <argument name="configPaths" xsi:type="array">
                <item name="general/imprint" xsi:type="array">
                    <item name="general/imprint/shop_name" xsi:type="string">1</item>
                    <item name="general/imprint/company_first" xsi:type="string">1</item>
                    <item name="general/imprint/company_second" xsi:type="string">1</item>
                    <item name="general/imprint/street" xsi:type="string">1</item>
                    <item name="general/imprint/zip" xsi:type="string">1</item>
                    <item name="general/imprint/city" xsi:type="string">1</item>
                    <item name="general/imprint/country" xsi:type="string">1</item>
                    <item name="general/imprint/telephone" xsi:type="string">1</item>
                    <item name="general/imprint/telephone_additional" xsi:type="string">1</item>
                    <item name="general/imprint/fax" xsi:type="string">1</item>
                    <item name="general/imprint/email" xsi:type="string">1</item>
                    <item name="general/imprint/web" xsi:type="string">1</item>
                    <item name="general/imprint/tax_number" xsi:type="string">1</item>
                    <item name="general/imprint/vat_id" xsi:type="string">1</item>
                    <item name="general/imprint/court" xsi:type="string">1</item>
                    <item name="general/imprint/financial_office" xsi:type="string">1</item>
                    <item name="general/imprint/ceo" xsi:type="string">1</item>
                    <item name="general/imprint/owner" xsi:type="string">1</item>
                    <item name="general/imprint/content_responsable_name" xsi:type="string">1</item>
                    <item name="general/imprint/content_responsable_address" xsi:type="string">1</item>
                    <item name="general/imprint/content_responsable_press_law" xsi:type="string">1</item>
                    <item name="general/imprint/register_number" xsi:type="string">1</item>
                    <item name="general/imprint/business_rules" xsi:type="string">1</item>
                    <item name="general/imprint/authority" xsi:type="string">1</item>
                    <item name="general/imprint/shareholdings" xsi:type="string">1</item>
                    <item name="general/imprint/editorial_concept" xsi:type="string">1</item>
                    <item name="general/imprint/bank_account_owner" xsi:type="string">1</item>
                    <item name="general/imprint/bank_account" xsi:type="string">1</item>
                    <item name="general/imprint/bank_code_number" xsi:type="string">1</item>
                    <item name="general/imprint/bank_name" xsi:type="string">1</item>
                    <item name="general/imprint/swift" xsi:type="string">1</item>
                    <item name="general/imprint/iban" xsi:type="string">1</item>
                    <item name="general/imprint/clearing" xsi:type="string">1</item>
                </item>
            </argument>
        </arguments>
    </type>
</config>
```

**Comment by mklooss** [comment]
an easy fix can be done in your composer projects with: this tool cweagans/composer-patches

**Comment by dominik-eller** [comment]
Can you explain this easy fix, please?

**Comment by sprankhub** [comment]
Could you please review / test https://github.com/firegento/firegento-magesetup2/pull/200? If someone tested / reviewed and the pipeline is fixed, I will merge and tag a new version `1.0.0`, which is then only compatible with Magento >= 2.4.3.

**Comment by Schrank** [comment]
> Can you explain this easy fix, please?

The composer plugin https://github.com/cweagans/composer-patches can apply patches. So what you can do with it is install Firegento_Magesetup and build a patch to change the di.xml, have this patch file either in your git or somewhere on http (which I don't recommend!) and then configure your composer.json to apply the patch AFTER composer install.

@arumcomputer

**Comment by sprankhub** [comment]
Released version [1.0.0](https://github.com/firegento/firegento-magesetup2/releases/tag/v1.0.0). Mind that due to the major version bump, you need to change the version constraint you require or simply remove and re-require the module.
